org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/hover/HoverContext.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2016, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ide.server.hover;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.ide.server.Document;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.util.ITextRegion;
+
+public class HoverContext {
```

Comment:
A generated `toString` may be handy for debugging

Reply:
am not sure if toString on the resource and element would be good

Reply:
Maybe `resource.getURI` is already helpful?

Reply:
done